File: map/lab2/src/repository/GradeValidator.java
package repository;

import domain.Grade;

import java.util.ArrayList;

public class GradeValidator implements Validator<Grade> {
    @Override
    public void validate(Grade gr) throws ValidationException {
        ArrayList<String> error_msg = new ArrayList<>();
        if (gr.getValue() > 10 || gr.getValue() < 1) {
            error_msg.add("Value isn't between 1 and 10.");
        }

        if (gr.getStId() < 1) {
            error_msg.add("The id of the student is invalid.");
        }

        if (gr.getPrId() < 1) {
            error_msg.add("The id of the project is invalid.");
        }

        if (!error_msg.isEmpty()) {
            throw new ValidationException(String.join("\n", error_msg));
        }
    }
}

File: map/lab7/lab7/Domain/GradeValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace lab7.Domain
{
    public class GradeValidator : IValidator<Grade>
    {
        public void Validate(Grade gr)
        {
            var errorMsg = new List<string>();
            if (gr.StId < 1) {
                errorMsg.Add("The id of the student is invalid/not found.");
            }

            if (gr.PrId < 1) {
                errorMsg.Add("The id of the project is invalid/not found.");
            }

            if (gr.Value > 10 || gr.Value < 1) {
                errorMsg.Add("Value isn't between 1 and 10.");
            }

            if (gr.InWeek > 14 || gr.InWeek < 1) {
                errorMsg.Add("Week isn't between 1 and 14.");
            }

            if (errorMsg.Any()) {
                throw new ValidationException(String.Join("\n", errorMsg));
            }
        }
    }
}